src/v/serde/parquet/column_writer.cc

```diff
@@ -176,19 +210,50 @@ class buffered_column_writer final : public column_writer::impl {
         full_page_data.append(std::move(encoded_rep_levels));
         full_page_data.append(std::move(encoded_def_levels));
         full_page_data.append(std::move(encoded_data));
-        _stats.reset();
-        co_return data_page{
+        _page_stats.reset();
+        _current_buffer_size = 0;
+        _pages.push_back(data_page{
           .header = std::move(header),
           .serialized_header_size = header_size,
           .serialized = std::move(full_page_data),
+        });
+    }
+
+    ss::future<flushed_pages> flush_pages() override {
+        if (_current_buffer_size > 0) {
+            co_await flush_page();
+        }
+
+        statistics full_stats{
+          .null_count = _total_stats.null_count(),
+          .max = {},
+          .min = {},
+        };
+        if (auto max = _total_stats.take_max()) {
+            full_stats.max.emplace(
+              /*value=*/encode_for_stats(*max),
+              /*is_exact=*/true);
+        }
+        if (auto min = _total_stats.take_min()) {
+            full_stats.min.emplace(
+              /*value=*/encode_for_stats(*min),
+              /*is_exact=*/true);
+        }
+        _total_stats.reset();
+        co_return flushed_pages{
+          .pages = std::exchange(_pages, {}),
+          .stats = std::move(full_stats),
         };
     }
 
 private:
-    column_stats_collector<value_type, comparator> _stats;
+    column_stats_collector<value_type, comparator> _page_stats;
+    column_stats_collector<value_type, comparator> _total_stats;
```

Comment:
nit: might be nice to comment that _total_stats isn't an on-going collection of stats, and rather that it only gets updated as pages get flushed. Maybe also worth mentioning that _page_stats is for the current page, not yet flushed in _pages?

Reply:
I don't feel strongly about any of them, but maybe _page_stats could be _current_page_stats, _total_stats could be _flushed_stats, and _pages could be _flushed_pages?

Reply:
I like it. Done!